tests/rng.ts

```diff
@@ -2,7 +2,7 @@ export function rng(max:Value, min:Value, state:Value) {
     unlock();
     let r = max - min + 1;
     if (state <= 0) {
-        state = factionItemAmount("metalore");
+        [,state] = factionItemAmount("metalore");
```

Comment:
This is why I didn't use multiple returns, this just feels awkward to me.
Do you have any example of something you're trying to do that doesn't work without the multi-return?
Maybe we could use aliases to provide both options for instructions that need it?

Reply:
I tried to make a behavior similar to this, where each output node is its own branch while also being able to refer to the value(s) returned by the instruction:

![image](https://github.com/ribrdb/desynced-tools/assets/65694696/2a2892c1-1a7d-4b25-afc4-62c40d0ee8dc)

With this change it would look like:

```typescript
    let [branch, nearest] = selectNearest(v, signal);
    switch(branch) {
        case "A":
            notify("A is closest", nearest);
            break;
        case "B":
            notify("B is closest", nearest);
            break;
        default:
            notify("Default branch");
            break;
    }
```

I agree that this syntax is not ideal, especially not for simple instructions who only return a value for their "truthy" branch.

Reply:
Another issue is the current typing system would prevent a user for writing:

![image](https://github.com/ribrdb/desynced-tools/assets/65694696/54b38e3e-8208-490e-9405-4e7cca3e76d1)

```typescript
signal = firstInventoryItem();
```

This is because the type definition says `firstInventoryItem(): Value | undefined` and so undefined cannot be asigned to signal.

The game is not at all strict about which branch returns a value or whether branches even represent truthy, falsey, presence or absence of variables. Trying to express it as such in typescript seems restrictive.

Reply:
I see, that explains some things. I've noticed things compiled anyway, but functional type checking is nice 😄 

I've checked all other instructions that currently exist in game and it seems that Select Nearest is the only one that has more than 2 branches while also returning a value.

However I do feel like it's important to point out that the game engine allows this, and it's not unfathomable more such instructions will show up in the future. And there's also mods.

Perhaps this new syntax should be opt-in on a case-by-case basis for aliases?

Reply:
> Perhaps this new syntax should be opt-in on a case-by-case basis for aliases?

Yeah, that definitely seems fine. Although as new instructions come up I'd want to think about what's the most natural way to use the instruction in JS. I don't really have any idea how to support instructions from mods.

Reply:
I'm just realizing how weird "Select Nearest" is. It also appears to be the only instruction that has optional "exec" args, and changes it's behavior based on whether or not you use them.

Reply:
I've made the behavior opt-in and opted in a new alias for selectNearest